packages/serve/src/index.ts

```diff
@@ -12,6 +12,10 @@ import parseArgs from './parseArgs';
 export default function serve(...args: string[]): void {
     const cli = new WebpackCLI();
 
+    // add WEBPACK_SERVE to main process and compilation environment
+    process.env['WEBPACK_SERVE'] = JSON.stringify(true);
+    args = [...args, '--env', 'process.env.WEBPACK_SERVE=true'];
```

Comment:
I think it is unsafe, why do you use `process.env['WEBPACK_SERVE'] = true` in code?

Reply:
In the reproducible repo of the issue, it was expected to access this variable through `process.env` directly in webpack config. W/O recieving it from `env` argument. To support this we would need to update it directly in process.

Reply:
Should I enforce use of `env` argument in webpack configs?

Reply:
hm, I think we need small discussion here, in webpack-dev-server we setup process.env['WEBPACK_SERVER'], but I found it some weird, so we need to decide how we will do it better:
- use old approach - just set `process.env['WEBPACK_SERVER']`
- passing `serve` in `env` flags by default

Reply:
> Should I enforce use of env argument in webpack configs?

Can you clarify?

Reply:
Both have their pros and cons.
1. Old approach: The variable is accessible by webpack config globally. But in this scenario, it would not be available outside webpack config. As it is old behaviour, people would be easily able to migrate for this breaking change.

2. New approach: The variable is not accessible by webpack config globally, use of `env` argument is required by webpack configuration. Plus point is, we would be able to access the variable outside of configuration as well. Con would be that migration of configuration would now be difficult as compared to previous approach.

P.S.
I have done both in this PR for now.

Reply:
> > Should I enforce use of env argument in webpack configs?
> 
> Can you clarify?

Yes, I mean access variable like this in webpack configuration.
```
module.exports = (env) => {
  env.WEBPACK_SERVE
  return {...};
}
```

Reply:
Honestly I found `env` is better solution

Reply:
Okay, will enforce the `env` argument then 😃.

Reply:
Done.